mopidy_listenbrainz/frontend.py

```diff
@@ -15,14 +19,135 @@ class ListenbrainzFrontend(pykka.ThreadingActor, CoreListener):
     def __init__(self, config, core):
         super().__init__()
         self.config = config
+        self.library = core.library
+        self.playlists = core.playlists
+        self.playlists_update_timer = None
 
     def on_start(self):
         self.lb = Listenbrainz(
             self.config["listenbrainz"]["token"],
             self.config["listenbrainz"]["url"],
+            self.config["proxy"],
         )
         logger.debug("Listenbrainz token valid!")
 
+        if self.config["listenbrainz"].get("import_playlists", False):
+            self.import_playlists()
+
+    def on_stop(self):
+        if self.playlists_update_timer:
+            self.playlists_update_timer.cancel()
+
+    def import_playlists(self) -> None:
+        logger.info("Import of ListenBrainz playlists")
+
+        import_count = 0
+        playlist_datas = self.lb.list_playlists_created_for_user()
+        logger.debug(f"Found {len(playlist_datas)} playlists to import")
+
+        existing_playlists = self.playlists.as_list().get()
+        recommendation_playlist_uri_prefix = (
+            "listenbrainz:playlist:recommendation"
+        )
+        filtered_existing_playlists = dict(
+            [
+                (ref.uri, ref)
+                for ref in existing_playlists
+                if ref.uri.startswith(recommendation_playlist_uri_prefix)
+            ]
+        )
+
+        for playlist_data in playlist_datas:
+            source = playlist_data.playlist_id
+            playlist_uri = f"{recommendation_playlist_uri_prefix}:{playlist_data.playlist_id}"
+            tracks = self._collect_playlist_tracks(playlist_data)
+
+            if len(tracks) == 0:
+                logger.debug(
+                    f"Skipping import of playlist with no known track for {source!r}"
+                )
+                continue
+
+            if playlist_uri in filtered_existing_playlists:
+                filtered_existing_playlists.pop(playlist_uri)
+                # must pop since filtered_existing_playlists will
+                # finally be deleted
+
+                logger.debug(f"Already known playlist {playlist_uri}")
+                # maybe there're new tracks in Mopidy's database...
```

Comment:
Do you mean that the playlist has been updated since import? Do the playlists change?

Reply:
From my experience, the playlists don't change from Listenbrainz backend point of view (but can't find the information in their documentation). From mopidy's point of view, they may change due to new tracks being imported in the library. The comment is related to that part.